### PR TITLE
feature/verification

### LINKED ISSUE #73: Add support for MOCBOT verification
Incorporate MOCBOT's Verification system into the API

### PR DESCRIPTION
**Pull request by MasterOfCubesAU**
### Added Verification support to MOCBOT API (now v1.1.0)
- Updated API doc to reflect new Verification system. See [API Docs](https://github.com/MasterOfCubesAU/mocbot-api/blob/feature/verification/docs/api.yml#L534)
- Added type definitions for new Verification objects
- Added testing and implemented:
  | Route | Description|
  |--------|--------------|
  | PUT /verification/guild_id/user_id | Puts a user into verification, optionally specifying if they should be locked down |
  | PATCH /verification/guild_id/user_id | Updates an existing user to be placed into lock down |
  | DELETE  /verification/guild_id/user_id | Removes an existing user from verification. |
  
Closes #73

### CODE REVIEW

File: src/verification/index.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  const userGuildID = await getUserGuildID(guildID, userID);
+  await DB.execute('DELETE FROM Verification WHERE UserGuildID = ?', [userGuildID]);
+  return {};
+}
+
+/**
+ * Adds a user from a given guild into verification. Putting this user into lock down immediately is possible
+ * @param {bigint | number} userID The User ID to add to verification
+ * @param {bigint | number} guildID The guild ID the user belongs to
+ * @param {LockdownInput} data the new lock down data to add to the user
```
Comment (sam1357) [inline src/verification/index.ts:52]:
```suggestion
 * @param {LockdownInput} the new lock down data to add to the user
```

File: src/verification/index.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * @throws {createErrors<400>} if provided lock down data is not complete or data is empty
+ * @throws {createErrors<404>} User/Guild ID not found
+ * @throws {createErrors<409>} User already in verification
```
Comment (sam1357) [inline src/verification/index.ts:55]:
```suggestion
 
```

File: docs/api.yml [hunk 3]
```diff
@@ -529,6 +531,104 @@ paths:
           $ref: "#/components/responses/NotAuthorized"
         "404":
           description: Warning ID not found
+  /verification/{guild_id}/{user_id}:
+    post:
```
Comment (sam1357) [inline docs/api.yml:535]: Post Verification should take in optional routes of MessageID and ChannelID

File: docs/api.yml [hunk 4]
```diff
@@ -645,5 +745,32 @@ components:
...
+        JoinTime:
+          type: integer
+          format: int64
+          description: The Unix timestamp at which the user was placed in Verification, in whole number of seconds
+          example: 1671727778
+    VerificationUpdate:
+      type: object
+      properties:
+        MessageID:
+          type: integer
```
Comment (sam1357) [inline docs/api.yml:768]:
```suggestion
          type: string
```

File: docs/api.yml [hunk 5]
```diff
@@ -645,5 +745,32 @@ components:
...
+          description: The ID of the lock down approval message
+          example: 1056206377569222656
+        ChannelID:
+          type: integer
```
Comment (sam1357) [inline docs/api.yml:772]:
```suggestion
          type: string
```

**Review by sam1357 — CHANGES_REQUESTED** [review]

File: docs/api.yml (hunk 4 above)
Comment (MasterOfCubesAU) [reply docs/api.yml:768]: Not necessary

File: docs/api.yml (hunk 5 above)
Comment (MasterOfCubesAU) [reply docs/api.yml:772]: Not necessary

**Review by sam1357 — APPROVED** [review]

**Pull request merged**